feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio_nl80211.c

```diff
@@ -227,6 +233,26 @@ static void nl80211_del_station(struct nlattr **tb, char *ifname)
 	_nl80211_del_station(sta);
 }
 
+static void remove_stale_station(char *ifname)
+{
+	struct wifi_station *sta = NULL, *tmp = NULL;
+	char mac[ETH_ALEN * 3];
+	time_t timeout = 0, t = 0;
+
+	avl_for_each_element_safe(&sta_tree, sta, avl, tmp) {
+		timeout = sta->last_update + (EVSCHED_SEC(STA_POLL_INTERVAL*2));
+		t = time(NULL);
+		if (t > timeout) {
+			print_mac(mac, sta->addr);
+			LOGN("%s: Remove stale STA %s %ld %ld %ld", __func__, mac,
+					t, timeout, sta->last_update);
+			vif_del_sta_rate_rule(sta->addr, ifname);
```

Comment:
This sequence of three calls is also made in nl80211_del_station().  Maybe it should be a little helper function.  If more operations are later tied to this data structure, then we eliminate the risk of them only being called in one code path.

Reply:
Done. Moved to _nl80211_del_station.